spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-deadLetterTopic-sample/src/main/java/com/example/SinkExample.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import java.util.function.Consumer;
+
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.cloud.spring.pubsub.support.BasicAcknowledgeablePubsubMessage;
+import com.google.cloud.spring.pubsub.support.GcpPubSubHeaders;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * Example of a sink for the sample app.
+ *
+ * @author Travis Tomsu
+ */
+@Configuration
+@Slf4j
+public class SinkExample {
+
+	@Bean
+	public Consumer<Message<UserMessage>> logUserMessage() {
```

Comment:
Should this also be a `Consumer<UserMessage>` like the below `deadLetterMessages()` for consistency? Or does one way only take the wrapped message object?

Reply:
Unfortunately no, because I need the ability to `nack()` the message. I've added a utility method that I'd like to get your opinion on for more reliably getting an object I can nack - though I'm not quite sure how this interacts with the acking/nacking happening within the framework. Maybe @elefeint would have some insight.